app/master/build.py

```diff
@@ -461,11 +462,19 @@ def _delete_temporary_build_artifact_files(self):
         ONLY call this method after _create_build_artifact() has completed. Otherwise we have lost the build results.
         """
         build_result_dir = self._build_results_dir()
+        self._logger.info('Deleting build artifact files for {} in {}.', self._build_id, build_result_dir)
         for path in os.listdir(build_result_dir):
             # The build result tar-ball is also stored in this same directory, so we must not delete it.
             if path == BuildArtifact.ARTIFACT_FILE_NAME:
                 continue
-            app.util.fs.async_delete(os.path.join(build_result_dir, path))
+            full_path = os.path.join(build_result_dir, path)
+            # Do NOT use app.util.fs.async_delete() here. That call will generate a temp directory for every
+            # atom, which can be in the thousands per build, and can lead to running up against the ulimit -Hn.
+            if os.path.isdir:
+                shutil.rmtree(full_path, ignore_errors=True)
+            else:
+                os.remove(full_path)
+        self._logger.info('Completed deleting artifact files for {}.', self._build_id)
```

Comment:
How do you feel about instead of a started and completed log if we just measured the time it takes and log that?